acp/identity.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package acp
+
+import (
+	cosmosSecp256k1 "github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"github.com/sourcenetwork/immutable"
+)
+
+// NoIdentity specifies an anonymous actor.
+var NoIdentity = immutable.None[Identity]()
+
+// Identity uniquely identifies an acp actor.
+type Identity interface {
+	// PublicKey returns the public key of the identity.
+	PublicKey() *secp256k1.PublicKey
+	// Address returns the bech32 address of the identity.
+	Address() string
+}
+
+var _ (Identity) = (*PublicKeyIdentity)(nil)
+
+// PublicKeyIdentity is an identity with only a public key.
+type PublicKeyIdentity struct {
```

Comment:
question: I can't spot where this is used, is it dead code?

Reply:
It's a client facing utility, but will be removed when I merge in Shahzad's identity changes.

Reply:
What will replace it?  Is there any value to it living for just the (short?) interval between merging this PR and Shahzad's changes?

I'm asking partly because SourceHub ACP needs stuff to be signed (client-side), which requires the private key.

Reply:
> What will replace it?

Something like this

```go
type Identity struct {
	// PublicKey is the identity public key.
	PublicKey *secp256k1.PublicKey
	// PrivateKey is the identity private key.
	PrivateKey *secp256k1.PrivateKey
	// Address is the identity address.
	Address string
}
```

> Is there any value to it living for just the (short?) interval between merging this PR and Shahzad's changes?

It's a quick change and Shahzad will most likely want this done before approving.